doc/en/multitest.rst

```diff
@@ -1024,11 +1026,18 @@ for sample implementation.
       """
       number_list = kwargs['number_list']
 
-      return '{func_name}__numbers_{joined_list}__result_{expected}'.format(
-          func_name=func_name,
-          joined_list='_'.join(map(str, number_list)),
-          expected=kwargs['expected']
-      )
+      if kwargs.get('for_display', False):
```

Comment:
I think this is over-complicating. Surely the custom name generator can just return a single human-readable name? Why does it need to be a valid python identifier?

Reply:
the name_func is used for generating names for parametrized testcases. Testplan will check if the generated name is not a valid python identifier then a warning message is printed, and testplan will simply reset the name with suffix. So user can know that their customized naming function does not work well. So, this name_func is designed to generate function name, not human readable display name. I also think that this is over-complicating, it is better to use another default name_func for generating hunan-readable names.

The question is: should we provide another another argument to let user customize the name_func for display name? if we should, that seems a little complicated, if we do not, then an unwanted name can be displayed such as "My test function (arg1: 1, arg2: [{'a': 'b'}, {'c': 'd'} ...])"

Reply:
Does not seem necessary to me at least for now. The idea is if user specify his own name_func, then he takes full control. I do not think it is super difficult to come up with a pattern that is both human-readable && python identifier valid.

Reply:
Why does the name need to be a valid python identifier?

I agree it's probably not needed for now.

Reply:
we have decided to add display name, so the name_func is now used for generating display name for parametrized testcases. and our default name func should be improved a little, because most time the string format of parameter is not simple positive integer or letters, and they will be renamed again.